<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>R: Representation and manipulation of short genomic alignments</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<link rel="stylesheet" type="text/css" href="R.css">
</head><body>
<h1> Representation and manipulation of short genomic alignments
<img class="toplogo" src="../../../doc/html/logo.jpg" alt="[R logo]">
</h1>
<hr>
<div align="center">
<a href="../../../doc/html/packages.html"><img src="../../../doc/html/left.jpg" alt="[Up]" width="30" height="30" border="0"></a>
<a href="../../../doc/html/index.html"><img src="../../../doc/html/up.jpg" alt="[Top]" width="30" height="30" border="0"></a>
</div><h2>Documentation for package &lsquo;GenomicAlignments&rsquo; version 1.2.2</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../doc/index.html">User guides, package vignettes and other documentation.</a></li>
<li><a href="../NEWS">Package NEWS</a>.</li>
</ul>

<h2>Help Pages</h2>


<p align="center">
<a href="#A">A</a>
<a href="#C">C</a>
<a href="#E">E</a>
<a href="#F">F</a>
<a href="#G">G</a>
<a href="#I">I</a>
<a href="#J">J</a>
<a href="#L">L</a>
<a href="#M">M</a>
<a href="#N">N</a>
<a href="#O">O</a>
<a href="#P">P</a>
<a href="#Q">Q</a>
<a href="#R">R</a>
<a href="#S">S</a>
<a href="#U">U</a>
<a href="#V">V</a>
<a href="#W">W</a>
<a href="#misc">misc</a>
</p>


<h2><a name="A">-- A --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="stackStringsFromBam.html">alphabetFrequencyFromBam</a></td>
<td>Stack the read sequences stored in a BAM file on a region of interest</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">as.data.frame-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">as.data.frame-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">as.data.frame.OverlapEncodings</a></td>
<td>OverlapEncodings objects</td></tr>
</table>

<h2><a name="C">-- C --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="GAlignmentPairs-class.html">c-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">c-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">c-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GappedReads-class.html">c-method</a></td>
<td>(Legacy) GappedReads objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">cigar</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">cigar-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">cigar-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigar-utils</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarNarrow</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarOpTable</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarQNarrow</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarRangesAlongPairwiseSpace</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarRangesAlongQuerySpace</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarRangesAlongReferenceSpace</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarToRleList</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarWidthAlongPairwiseSpace</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarWidthAlongQuerySpace</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">cigarWidthAlongReferenceSpace</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">CIGAR_OPS</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">class:GAlignmentPairs</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">class:GAlignments</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">class:GAlignmentsList</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GappedReads-class.html">class:GappedReads</a></td>
<td>(Legacy) GappedReads objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">class:OverlapEncodings</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">coerce-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">coerce-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">coerce-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="findCompatibleOverlaps-methods.html">countCompatibleOverlaps</a></td>
<td>Finding hits between reads and transcripts that are _compatible_ with the splicing of the transcript</td></tr>
<tr><td width="25%"><a href="findMateAlignment.html">countDumpedAlignments</a></td>
<td>Pairing the elements of a GAlignments object</td></tr>
<tr><td width="25%"><a href="findOverlaps-methods.html">countOverlaps</a></td>
<td>Finding overlapping genomic alignments</td></tr>
<tr><td width="25%"><a href="findOverlaps-methods.html">countOverlaps-method</a></td>
<td>Finding overlapping genomic alignments</td></tr>
<tr><td width="25%"><a href="coverage-methods.html">coverage</a></td>
<td>Coverage of a GAlignments or GAlignmentPairs object</td></tr>
<tr><td width="25%"><a href="coverage-methods.html">coverage-method</a></td>
<td>Coverage of a GAlignments or GAlignmentPairs object</td></tr>
<tr><td width="25%"><a href="coverage-methods.html">coverage-methods</a></td>
<td>Coverage of a GAlignments or GAlignmentPairs object</td></tr>
</table>

<h2><a name="E">-- E --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="GAlignmentsList-class.html">elementMetadata-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">elementMetadata&lt;--method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">elementMetadata&lt;--method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">elementMetadata&lt;--method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">encodeOverlaps</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">encodeOverlaps-method</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">encodeOverlaps-methods</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">encodeOverlaps1</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">encoding-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">end-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">end-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">explodeCigarOpLengths</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">explodeCigarOps</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">extractAlignmentRangesOnReference</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">extractQueryStartInTranscript</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">extractSkippedExonRanks</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">extractSkippedExonRanks-method</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">extractSpannedExonRanks</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">extractSpannedExonRanks-method</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">extractSteppedExonRanks</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">extractSteppedExonRanks-method</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
</table>

<h2><a name="F">-- F --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="findCompatibleOverlaps-methods.html">findCompatibleOverlaps</a></td>
<td>Finding hits between reads and transcripts that are _compatible_ with the splicing of the transcript</td></tr>
<tr><td width="25%"><a href="findCompatibleOverlaps-methods.html">findCompatibleOverlaps-method</a></td>
<td>Finding hits between reads and transcripts that are _compatible_ with the splicing of the transcript</td></tr>
<tr><td width="25%"><a href="findCompatibleOverlaps-methods.html">findCompatibleOverlaps-methods</a></td>
<td>Finding hits between reads and transcripts that are _compatible_ with the splicing of the transcript</td></tr>
<tr><td width="25%"><a href="findMateAlignment.html">findMateAlignment</a></td>
<td>Pairing the elements of a GAlignments object</td></tr>
<tr><td width="25%"><a href="findMateAlignment.html">findMateAlignment2</a></td>
<td>Pairing the elements of a GAlignments object</td></tr>
<tr><td width="25%"><a href="findOverlaps-methods.html">findOverlaps</a></td>
<td>Finding overlapping genomic alignments</td></tr>
<tr><td width="25%"><a href="findOverlaps-methods.html">findOverlaps-method</a></td>
<td>Finding overlapping genomic alignments</td></tr>
<tr><td width="25%"><a href="findOverlaps-methods.html">findOverlaps-methods</a></td>
<td>Finding overlapping genomic alignments</td></tr>
<tr><td width="25%"><a href="findSpliceOverlaps-methods.html">findSpliceOverlaps</a></td>
<td>Classify ranges (reads) as compatible with existing genomic annotations or as having novel splice events</td></tr>
<tr><td width="25%"><a href="findSpliceOverlaps-methods.html">findSpliceOverlaps-method</a></td>
<td>Classify ranges (reads) as compatible with existing genomic annotations or as having novel splice events</td></tr>
<tr><td width="25%"><a href="findSpliceOverlaps-methods.html">findSpliceOverlaps-methods</a></td>
<td>Classify ranges (reads) as compatible with existing genomic annotations or as having novel splice events</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">first</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">first-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">flippedQuery</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">flippedQuery-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">flipQuery</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="findMateAlignment.html">flushDumpedAlignments</a></td>
<td>Pairing the elements of a GAlignments object</td></tr>
</table>

<h2><a name="G">-- G --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="GAlignmentPairs-class.html">GAlignmentPairs</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">GAlignmentPairs-class</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">GAlignments</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">GAlignments-class</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">GAlignmentsList</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">GAlignmentsList-class</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GappedReads-class.html">GappedReads</a></td>
<td>(Legacy) GappedReads objects</td></tr>
<tr><td width="25%"><a href="GappedReads-class.html">GappedReads-class</a></td>
<td>(Legacy) GappedReads objects</td></tr>
<tr><td width="25%"><a href="findMateAlignment.html">getDumpedAlignments</a></td>
<td>Pairing the elements of a GAlignments object</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">granges-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">granges-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">granges-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">grglist-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">grglist-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">grglist-method</a></td>
<td>GAlignmentsList objects</td></tr>
</table>

<h2><a name="I">-- I --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="summarizeOverlaps-methods.html">IntersectionNotEmpty</a></td>
<td>Perform overlap queries between reads and genomic features</td></tr>
<tr><td width="25%"><a href="summarizeOverlaps-methods.html">IntersectionStrict</a></td>
<td>Perform overlap queries between reads and genomic features</td></tr>
<tr><td width="25%"><a href="intra-range-methods.html">intra-range-methods</a></td>
<td>Intra range transformations of a GAlignments or GAlignmentsList object</td></tr>
<tr><td width="25%"><a href="junctions-methods.html">introns</a></td>
<td>Extract junctions from genomic alignments</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">isCompatibleWithSkippedExons</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">isCompatibleWithSkippedExons-method</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">isCompatibleWithSplicing</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="encodeOverlaps-methods.html">isCompatibleWithSplicing-method</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">isProperPair</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">isProperPair-method</a></td>
<td>GAlignmentPairs objects</td></tr>
</table>

<h2><a name="J">-- J --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="junctions-methods.html">junctions</a></td>
<td>Extract junctions from genomic alignments</td></tr>
<tr><td width="25%"><a href="junctions-methods.html">junctions-method</a></td>
<td>Extract junctions from genomic alignments</td></tr>
<tr><td width="25%"><a href="junctions-methods.html">junctions-methods</a></td>
<td>Extract junctions from genomic alignments</td></tr>
</table>

<h2><a name="L">-- L --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="GAlignmentPairs-class.html">last</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">last-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">left</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">left-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Lencoding</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Lencoding-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">length-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">length-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">length-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">levels-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">levels.OverlapEncodings</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Lngap</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Lnjunc</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Lnjunc-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Loffset</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Loffset-method</a></td>
<td>OverlapEncodings objects</td></tr>
</table>

<h2><a name="M">-- M --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="findMateAlignment.html">makeGAlignmentPairs</a></td>
<td>Pairing the elements of a GAlignments object</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">makeGAlignmentsListFromFeatureFragments</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="map-methods.html">map-method</a></td>
<td>Mapping ranges between sequences</td></tr>
<tr><td width="25%"><a href="map-methods.html">map-methods</a></td>
<td>Mapping ranges between sequences</td></tr>
<tr><td width="25%"><a href="mapCoords-methods.html">mapCoords-method</a></td>
<td>Mapping ranges between sequences</td></tr>
<tr><td width="25%"><a href="mapCoords-methods.html">mapCoords-methods</a></td>
<td>Mapping ranges between sequences</td></tr>
</table>

<h2><a name="N">-- N --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="GAlignmentPairs-class.html">names-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">names-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">names-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">names&lt;--method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">names&lt;--method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">names&lt;--method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="intra-range-methods.html">narrow</a></td>
<td>Intra range transformations of a GAlignments or GAlignmentsList object</td></tr>
<tr><td width="25%"><a href="intra-range-methods.html">narrow-method</a></td>
<td>Intra range transformations of a GAlignments or GAlignmentsList object</td></tr>
<tr><td width="25%"><a href="junctions-methods.html">NATURAL_INTRON_MOTIFS</a></td>
<td>Extract junctions from genomic alignments</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">ngap-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">ngap-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">ngap-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">ngap-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">njunc</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">njunc-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">njunc-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">njunc-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">njunc-method</a></td>
<td>OverlapEncodings objects</td></tr>
</table>

<h2><a name="O">-- O --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="OverlapEncodings-class.html">OverlapEncodings</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">OverlapEncodings-class</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="findOverlaps-methods.html">overlapsAny</a></td>
<td>Finding overlapping genomic alignments</td></tr>
<tr><td width="25%"><a href="findOverlaps-methods.html">overlapsAny-method</a></td>
<td>Finding overlapping genomic alignments</td></tr>
</table>

<h2><a name="P">-- P --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="pileLettersAt.html">pileLettersAt</a></td>
<td>Pile the letters of a set of aligned reads on top of a set of individual genomic positions</td></tr>
<tr><td width="25%"><a href="setops-methods.html">pintersect-method</a></td>
<td>Set operations on GAlignments objects</td></tr>
</table>

<h2><a name="Q">-- Q --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="intra-range-methods.html">qnarrow</a></td>
<td>Intra range transformations of a GAlignments or GAlignmentsList object</td></tr>
<tr><td width="25%"><a href="intra-range-methods.html">qnarrow-method</a></td>
<td>Intra range transformations of a GAlignments or GAlignmentsList object</td></tr>
<tr><td width="25%"><a href="GappedReads-class.html">qseq</a></td>
<td>(Legacy) GappedReads objects</td></tr>
<tr><td width="25%"><a href="GappedReads-class.html">qseq-method</a></td>
<td>(Legacy) GappedReads objects</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">queryLoc2refLoc</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="cigar-utils.html">queryLocs2refLocs</a></td>
<td>CIGAR utility functions</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">qwidth</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">qwidth-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">qwidth-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GappedReads-class.html">qwidth-method</a></td>
<td>(Legacy) GappedReads objects</td></tr>
</table>

<h2><a name="R">-- R --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="GAlignments-class.html">ranges-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">ranges-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGAlignmentPairs</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGAlignmentPairsFromBam</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGAlignmentPairsFromBam-method</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGAlignments</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGAlignmentsFromBam</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGAlignmentsFromBam-method</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGAlignmentsList</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGAlignmentsListFromBam</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGAlignmentsListFromBam-method</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGappedReads</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGappedReadsFromBam</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="readGAlignments.html">readGappedReadsFromBam-method</a></td>
<td>Reading genomic alignments from a file</td></tr>
<tr><td width="25%"><a href="junctions-methods.html">readSTARJunctions</a></td>
<td>Extract junctions from genomic alignments</td></tr>
<tr><td width="25%"><a href="junctions-methods.html">readTopHatJunctions</a></td>
<td>Extract junctions from genomic alignments</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">relistToClass-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Rencoding</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Rencoding-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">rglist-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">rglist-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">right</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">right-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">rname</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">rname-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">rname-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">rname&lt;-</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">rname&lt;--method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">rname&lt;--method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Rngap</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Rnjunc</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Rnjunc-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Roffset</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">Roffset-method</a></td>
<td>OverlapEncodings objects</td></tr>
</table>

<h2><a name="S">-- S --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="encodeOverlaps-methods.html">selectEncodingWithCompatibleStrand</a></td>
<td>Encode the overlaps between RNA-seq reads and the transcripts of a gene model</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">seqinfo-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">seqinfo-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">seqinfo-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">seqinfo&lt;--method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">seqinfo&lt;--method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">seqinfo&lt;--method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">seqlevelsInUse-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">seqnames-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">seqnames-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">seqnames-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">seqnames&lt;--method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">seqnames&lt;--method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="sequenceLayer.html">sequenceLayer</a></td>
<td>Lay read sequences alongside the reference space, using their CIGARs</td></tr>
<tr><td width="25%"><a href="setops-methods.html">setops-methods</a></td>
<td>Set operations on GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">show-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">show-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">show-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="OverlapEncodings-class.html">show-method</a></td>
<td>OverlapEncodings objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">splitAsListReturnedClass-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="stackStringsFromBam.html">stackStringsFromBam</a></td>
<td>Stack the read sequences stored in a BAM file on a region of interest</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">start-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">start-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">strand-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">strand-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">strand-method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">strand&lt;--method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">strand&lt;--method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">strand&lt;--method</a></td>
<td>GAlignmentsList objects</td></tr>
<tr><td width="25%"><a href="findOverlaps-methods.html">subsetByOverlaps</a></td>
<td>Finding overlapping genomic alignments</td></tr>
<tr><td width="25%"><a href="findOverlaps-methods.html">subsetByOverlaps-method</a></td>
<td>Finding overlapping genomic alignments</td></tr>
<tr><td width="25%"><a href="junctions-methods.html">summarizeJunctions</a></td>
<td>Extract junctions from genomic alignments</td></tr>
<tr><td width="25%"><a href="summarizeOverlaps-methods.html">summarizeOverlaps</a></td>
<td>Perform overlap queries between reads and genomic features</td></tr>
<tr><td width="25%"><a href="summarizeOverlaps-methods.html">summarizeOverlaps-method</a></td>
<td>Perform overlap queries between reads and genomic features</td></tr>
<tr><td width="25%"><a href="summarizeOverlaps-methods.html">summarizeOverlaps-methods</a></td>
<td>Perform overlap queries between reads and genomic features</td></tr>
</table>

<h2><a name="U">-- U --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="summarizeOverlaps-methods.html">Union</a></td>
<td>Perform overlap queries between reads and genomic features</td></tr>
<tr><td width="25%"><a href="GAlignmentPairs-class.html">unlist-method</a></td>
<td>GAlignmentPairs objects</td></tr>
<tr><td width="25%"><a href="GAlignments-class.html">updateObject-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">updateObject-method</a></td>
<td>GAlignmentsList objects</td></tr>
</table>

<h2><a name="V">-- V --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="cigar-utils.html">validCigar</a></td>
<td>CIGAR utility functions</td></tr>
</table>

<h2><a name="W">-- W --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="GAlignments-class.html">width-method</a></td>
<td>GAlignments objects</td></tr>
<tr><td width="25%"><a href="GAlignmentsList-class.html">width-method</a></td>
<td>GAlignmentsList objects</td></tr>
</table>

<h2><a name="misc">-- misc --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="GAlignmentPairs-class.html">[[-method</a></td>
<td>GAlignmentPairs objects</td></tr>
</table>
</body></html>
