# Transposable Element Next Generation Sequencing  

Transposable elements (TEs) are sequences that can transpose or "jump" from position to position within genomes. In humans, this jumping mechanism has resulted in sequence diversity among individuals, including phenotypes such as genetic disease. With Gerton Lunter, we have developed genomics tools for TE-enriched next generation sequencing and detection of polymorphic TEs in human genomes.   

## About  
The method - described here [pdf](#) consists of  
  
1. Molecular genomics protocol for generating TE-enriched NGS libraries  
- detailed procedures [protocol](#)   
 
2. Computational pipeline for detecting TE insertions   
- get the source [code](https://github.com/ekviky/TE-NGS)  

TE calling algorithm consists of two principle steps:  
 (i) clustering on genomic coordinates  
 (ii) annotation via comparison to public and local TE databases  

## Implementation  
TE-NGS is implemented in R and distributed using [packrat](https://github.com/rstudio/packrat) for management of packages, dependencies, etc. The snapshot is built on R version 3.1.2.  

The R tool employs the following packages:  
- ShortRead for efficient containers to read/process bam files
- Rsamtools for bam manipulation  
- GenomicRanges for manipulation of genomic coordinates  
- Biostrings for fast implementation of regular expression pattern matching  
- data.table for binary search   

The R script requires the following arguments:
- seq_opt_table.txt # a file describing the experimental conditions per sample, eg sample ID, index, primer  
- path to bam directory # a directory containing bam(s) generated by TE-NGS workflow  
- cluster distance (bp) # minimum distance required between neighboring genomic clusters, 201 recommended  
- mode # one of [test,sample] indicating the run mode: test to generate TE calls on NA12878 bams provided in /test; sample for generating TE calls on new sample  

## Getting started  

Navigate to /distrib and unbundle the packrat tarball    
 
    > packrat::unbundle(bundle='scripts.tar.gz', where='/foo/TE-NGS/distrib')  

where /foo is wherever the TE-NGS repo lives locally 

Should see progress message:   

    Untarring 'scripts.tar.gz' in directory '/foo/TE-NGS/distrib'...  

    ...  

    Done! The project has been unbundled and restored at:  
    - "/foo/TE-NGS/distrib/scripts"  


Navigate to /scripts where packrat snapshot is built 
 
    $ cd scripts 

First time launching R, packrat will check the build of the local library of required packages and dependencies. Should see message, eg.  

    $ R
    ...
    Packrat mode on. Using library in directory:  
    - "/foo/TE-NGS/distrib/scripts/packrat/lib"  


Run the script in test mode: check that output as expected for test sample in /distrib/test, eg.   

    $  R --no-restore --no-save --no-readline --quiet < R_get_TE_calls_v0.1.r --args ../test/ ../test/seq_opt_table.txt 201 test  


Run the script to generate TE calls  
    
    $ R --no-restore --no-save --no-readline --quiet < R_get_TE_calls_v0.1.r --args /bamdir/ /bamdir/seq_opt_table.txt 201 sample  


## Distribution
Along with the source code several additional files are provided:   

The directory /annotations contains genomic interval flat files in build GRhs37 (hg19) for annotation of TE calls with known insertions.   

- polyTEdb - we compiled an extensive local database of known polymorhic TE insertions obtained from public databases and TE protocols (list references). Each TE is cross-referenced giving the number of sources, subfamily annotation in each source (for disambiguation), and primary source reference. 
```
    $ gunzip -c ../distrib/annotations/polyTEdb_window3flank600bp.interval.gz | more  
    chr1    645109  645710  1       ALU,    AluYa4_5,       phase3, 
    chr1    697981  698582  1       NA,     NA,     Witherspoon, 
    chr1    812282  812883  1       LINE1,  LINE1,  phase3, 
```

The directory /examples 

- NA12878_TE-NGS_calls_annotated_v0.1.bed - a flat file containing all TE-NGS calls made for NA12878 as described in manuscript.  


## Read more
- email
- other [projects](https://ekviky.github.io/about/)
