# Transposable Element Next Generation Sequencing  

Transposable elements (TEs) are sequences that can transpose or "jump" from position to position within genomes. In humans, this jumping mechanism has resulted in sequence diversity among individuals, including phenotypes such as genetic disease. With Gerton Lunter, we have developed genomics tools for TE-enriched next generation sequencing and detection of polymorphic TEs in human genomes.   

## About  
Why target TEs?  

1. **TEs are ubiquitous**: ~1/2 human genome is annotated as TE, and up to ~2/3 can be identified as ancient relics of TEs
2. **TEs are rare, but active**: 3 subfamilies account for vast majority (>95%) of polymorphic TEs in humans, but combined account for <0.2% of the genome
3. **TEs are tricky to sequence**: an estimated minimum of 140x coverage is needed to identify TEs reliably (>90% sensitivity) from whole genome sequencing 

We developed TE-NGS to overcome these challenges. TE-NGS is

- **High-throughput**: NGS sequencing-based
- **Comprehensive**: targets the 3 (L1HS, AluYa5/8, and AluYb8/9) most active TE subfamilies simultaneously
- **Practical**: assembles routine molecular genomics techniques 

The method - described here [pdf](#) consists of two parts
  
1. a molecular genomics protocol for generating TE-enriched NGS libraries  
- detailed procedures [protocol](#)   
 
2. a computational pipeline for detecting TE insertions   
- get the source [code](https://github.com/ekviky/TE-NGS)  

## Implementation  

The TE-NGS algorithm consists of two principle steps to identify TE calls:  
(i) clustering of reads based on genomic coordinates  
(ii) annotation of clusters via comparison to public and local TE databases  

TE-NGS is implemented in R and distributed using [packrat](https://github.com/rstudio/packrat) for management of packages, dependencies, etc. The snapshot is built on R version 3.1.2.  

The R tool employs the following packages:  
- ShortRead for efficient containers to read/process bam files
- Rsamtools for bam manipulation  
- GenomicRanges for manipulation of genomic coordinates  
- Biostrings for fast implementation of regular expression pattern matching  
- data.table for binary search   

The R script requires the following arguments:
- seq_opt_table.txt # a file describing the experimental conditions per sample, eg sample ID, index, primer  
```
$ cat seq_opt_table.txt
experiment	category	P1	P3	index
NA12878_A	element_A	"Alu_target","L1HS_target"	L1HS_nested	237
NA12878_C	element_C	"Alu_target","L1HS_target"	AluYb89_nested	277
```

- path to bam directory # a directory containing bam(s) generated by TE-NGS workflow  

- cluster distance (bp) # minimum distance required between neighboring genomic clusters [201 default]  

- mode # one of [test,sample] indicating the run mode
test mode generates TE calls on NA12878 bams provided in /test
sample mode generates TE calls on sample bams generated by TE-NGS workflow

## Getting started  

Navigate to /distrib and unbundle the packrat tarball    
 
    > packrat::unbundle(bundle='scripts.tar.gz', where='/foo/TE-NGS/distrib')  

where /foo points to wherever the TE-NGS repo lives locally 

Should expect to see a progress message:   

    Untarring 'scripts.tar.gz' in directory '/foo/TE-NGS/distrib'...  

    ...  

    Done! The project has been unbundled and restored at:  
    - "/foo/TE-NGS/distrib/scripts"  


Navigate to /scripts where packrat snapshot is built 
 
    $ cd scripts 

First time launching R, packrat will check the build of the local library of required packages and dependencies. Should see message, eg.  

    $ R
    ...
    Packrat mode on. Using library in directory:  
    - "/foo/TE-NGS/distrib/scripts/packrat/lib"  


Run the script in test mode: checks are performed to ensure that the output is as expected for test sample in /distrib/test, eg.   

    $  R --no-restore --no-save --no-readline --quiet < R_get_TE_calls_v0.1.r --args ../test/ ../test/seq_opt_table.txt 201 test  


Run the script to generate TE calls  
    
    $ R --no-restore --no-save --no-readline --quiet < R_get_TE_calls_v0.1.r --args /bamdir/ /bamdir/seq_opt_table.txt 201 sample  

## Troubleshooting

Ensure packrat is on  

    > packrat::on()

Check that packrat is up to date

    > packrat::status()

## Distribution
Along with the source code several additional files are provided:   

The directory /annotations contains genomic interval flat files in build GRhs37 (hg19) for annotation of TE calls with known insertions.   

- polyTEdb - we compiled an extensive local database of known polymorhic TE insertions obtained from public databases and TE protocols (list references). Each TE is cross-referenced giving the number of sources, subfamily annotation in each source (for disambiguation), and primary source reference. 
```
    $ gunzip -c polyTEdb_window3flank600bp.interval.gz | more  
    chr1    645109  645710  1       ALU,    AluYa4_5,       phase3, 
    chr1    697981  698582  1       NA,     NA,     Witherspoon, 
    chr1    812282  812883  1       LINE1,  LINE1,  phase3, 
```

The directory /examples 

- NA12878_TE-NGS_calls_annotated_v0.1.bed - a flat file containing all TE-NGS calls made for NA12878 as described in manuscript.  


## Read more
- email
- other [projects](https://ekviky.github.io/about/)
